tests/apps/demultiplex/test_index.py

```diff
@@ -101,9 +101,134 @@ def test_get_hamming_distance_index_1_different_prefixes():
     # when aligned to the left
     sequence_1: str = "GATXX"
     sequence_2: str = "GATTACA"
+    # WHEN getting the hamming distance between them in any order
+
+    # THEN the distance is equal to the number of different characters
+    assert get_hamming_distance_index_1(sequence_1=sequence_1, sequence_2=sequence_2) == 2
+    assert get_hamming_distance_index_1(sequence_1=sequence_2, sequence_2=sequence_1) == 2
+
+
+def test_get_hamming_distance_index_2_different_lengths_reverse_complement():
+    """Test that hamming distance between indexes with same suffix but different lengths is zero."""
+    # GIVEN two index_2 sequences with the same suffixes but different lengths
+    sequence_1: str = "GATTACA"
+    sequence_2: str = "XXGATTACA"
+
+    # WHEN getting the hamming distance between them in any order
+
+    # THEN the distance is zero
+    assert (
+        get_hamming_distance_index_2(
+            sequence_1=sequence_1, sequence_2=sequence_2, is_reverse_complement=True
+        )
+        == 0
+    )
+    assert (
+        get_hamming_distance_index_2(
+            sequence_1=sequence_2, sequence_2=sequence_1, is_reverse_complement=True
+        )
+        == 0
+    )
```

Comment:
Might be nice to assign the function return to a descriptive variable to it can be understood a bit easier what is being tested, hard to spot the that sequence 1 and 2 swapped positions

Reply:
I can't think of any better name, I will leave it as it is for now